src/wasm/wasm-s-parser.cpp

```diff
@@ -3507,7 +3512,9 @@ void SExpressionWasmBuilder::parseEvent(Element& s, bool preParseImport) {
   event->attribute = atoi(attrElem[1]->c_str());
 
   // Parse typeuse
-  i = parseTypeUse(s, i, event->sig);
+  HeapType eventType;
+  i = parseTypeUse(s, i, eventType);
+  event->sig = eventType.getSignature();
```

Comment:
This repeated pattern is slightly annoying, but I can't think of an obvious improvement...

Reply:
A (large) follow-up will change functions and events to store HeapTypes rather than signatures, so this particular pattern will go away.